### PR TITLE
Fixes for queue storage

### PR DESCRIPTION
**Pull request by elenasolomon**
1) fix for how the canonicalized headers are constructed: based on the documentation, each HTTP header name should be converted to lowercase and, any white space around the colon in the header should be trimmed.

2) fix the retrieve of the queue metadata
When trying to get the metadata for a queue with the following metadata:

{
    clientName: 'fast-azure-storage'
}

the response headers will look something like this:

Transfer-Encoding: chunked
x-ms-clientname: fast-azure-storage
Date: Fri, 16 Sep 2011 01:27:38 GMT
Server: Windows-Azure-Queue/1.0 Microsoft-HTTPAPI/2.0

Meaning that the result of the getMetadata will be:
{
    clientname: 'fast-azure-storage'
}

This happens because the 'https' library returns the HTTP headers with lowercase. The solution was to read the header name from response.rawHeaders.

### CODE REVIEW

File: lib/queue.js [hunk 1]
```diff
@@ -110,7 +110,9 @@ var authorizeWithSharedKey = function (method, path, query, headers) {
     var field = fields[i];
     var value = headers[field];
     if (value) {
-      stringToSign += '\n' + field + ':' + value;
+      // Convert each HTTP header to lower case and
+      // trim any white space around the colon in the header.
+      stringToSign += '\n' + field.toLowerCase() + ':' + value.trim();
```
Comment (jhford) [inline lib/queue.js]: is whitespace in the value an issue, or only at the beginning and end?  If having whitespace inside of the string is also an issue, we should probably test for whitespace in the trimmed string and throw an exception.

File: lib/queue.js [hunk 2]
```diff
@@ -650,9 +652,19 @@ Queue.prototype.getMetadata = function getMetadata(queue) {
...
       if (/x-ms-meta-/.test(field)) {
-        metadata[field.substr(10)] = res.headers[field];
+        // Metadata names must adhere to the naming rules for C# identifiers, which are case-insensitive,
+        // meaning that you can set,for example, a metadata with the following form:
+        // {
+        //    applicationName: 'fast-azure-blob-storage'
+        // }
+        // In order to return the original metadata name, the header names should be read from response.rawHeaders.
+        // That is because 'https' library returns the response headers with lowercase.
+        var key = res.rawHeaders[rawHeaderCounter * 2 - 2];
```
Comment (jhford) [inline lib/queue.js]: Sorry, I can't remember or find the structure of `response.rawHeaders`.  The comment is great, but could you include a brief explanation of why you're using the `rawHeaderCount * 2 - 2` value?

File: lib/queue.js (hunk 1 above)
Comment (elenasolomon) [inline lib/queue.js]: The whitespace in the value are not an issue.
